### ISSUE TITLE
Kube-dns to Core-dns migration fails 

### ISSUE THREAD

**Issue by mattmattox** [rancher/rancher#20205]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Deploy a custom cluster in Rancher v2.2.2 using the defaults ie using Kube-DNS
- Update the the cluster.yaml to change Kube-DNS to Core-DNS

**Result:**
When running Kubernetes 1.11.6 - kube-dns is removed and core is deploys but fails to come up
When running Kubernetes 1.13.5 - kube-dns is removed but doesn't deploy Core-DNS

**Other details that may be helpful:**
When running the following script https://github.com/salla2/deployment/tree/master/kubernetes the migration works fine. 

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.2.2
- Installation option (single install/HA): HA

**Comment by galal-hussein** [comment]
@mattmattox I can confirm that CoreDNS has a problem with Kubernetes 1.11, it looks like the image is not defined for this version, however I can't reproduce the problem at all with 1.13 for both rke standalone and in Rancher environment 

![coredns](https://user-images.githubusercontent.com/3309968/57807557-b0028900-7761-11e9-940f-9625f60bf95e.png)

**Comment by mattmattox** [comment]
@galal-hussein Did you build the cluster using RKE or a custom cluster?
Because I only saw this issue when using a custom cluster and making the change via the UI.

**Comment by galal-hussein** [comment]
@mattmattox Yes I did build a custom cluster, and I also made the change from the UI

**Comment by jambajaar** [comment]
@mattmattox  -- was the problem seen with v2.2.2 or in some rc build?

**Comment by mattmattox** [comment]
@galal-hussein So this environment was running v2.1.6 then was upgraded to v2.2.2 before running this upgrade.

**Comment by salla2** [comment]
@galal-hussein i was able to replace kube-dns with coredns on rancher 2.2.2 with k8s 13.5. Problem i'm facing is its not resolving external dns . When i changed back to kube-dns its working. Any insights?

**Comment by metal3d** [comment]
I found a weird solution.
- removed kube-dns remover job
- edit cluster and change something (e.g. etcd retention)

Save...

Deployment for CoreDNS starts.

So it seems that kube-dns-remover jobs breaks something

**Comment by CAChristopher** [comment]
Just ran into this updating from 1.13.5 to 1.14.8.  Kube-dns got stuck deleting and CoreDNS was never installed.  Luckily @metal3d's method of manually removing kubedns and the remover job and reinstalling worked.
